directives/ShouldBeAuthenticated.go

```diff
@@ -0,0 +1,27 @@
+package directives
+
+import (
+	"context"
+	"errors"
+
+	"github.com/99designs/gqlgen/graphql"
+	"github.com/PrameshKarki/event-management-golang/utils"
+)
+
+func ShouldBeAuthenticated() func(ctx context.Context, obj interface{}, next graphql.Resolver) (interface{}, error) {
+	return func(ctx context.Context, obj interface{}, next graphql.Resolver) (interface{}, error) {
+		// Get user if from context
+		user := ctx.Value("user")
+		if user != nil {
+			userID := user.(*utils.TokenMetadata).ID
+			if userID == "" {
+				return nil, errors.New("unauthorized")
+
+			} else {
+				return next(ctx)
+			}
+		} else {
+			return nil, errors.New("unauthorized")
```

Comment:
Since, this function is checking whether user is logged in or not, throwing `unauthenticated`  would make more sense since, `unauthorized` is normally related to not having permission for a resource

```suggestion
				return nil, errors.New("unauthenticated")

			} else {
				return next(ctx)
			}
		} else {
			return nil, errors.New("unauthenticated")
```